cpp/velox/operators/writer/VeloxParquetDatasource.h

```diff
@@ -49,8 +53,13 @@ class VeloxParquetDatasource final : public Datasource {
   VeloxParquetDatasource(
       const std::string& filePath,
       std::shared_ptr<facebook::velox::memory::MemoryPool> veloxPool,
+      std::shared_ptr<facebook::velox::memory::MemoryPool> s3SinkPool,
       std::shared_ptr<arrow::Schema> schema)
-      : Datasource(filePath, schema), filePath_(filePath), schema_(schema), pool_(std::move(veloxPool)) {}
+      : Datasource(filePath, schema),
+        filePath_(filePath),
+        schema_(schema),
+        pool_(std::move(veloxPool)),
+        s3SinkPool_(std::move(s3SinkPool)) {}
```

Comment:
This looks like adding the pool for all backends, including hdfs/local?

Reply:
Yes. But this pool is actually not new created and it's only used in S3.